<launch>

	<arg name="camera" default="kinect1"/>
	<arg name="online" default="true"/>
	<arg name="bus" default="1"/>
	<arg name="filename"/>

	<group if="$(arg online)">
		<param name="use_sim_time" value="false" />
		<include file="$(find openni_launch)/launch/openni.launch">
			<arg name="camera" value="$(arg camera)" />
			<arg name="device_id" value="$(arg bus)@0" />
			<arg name="depth_registration" value="true" />
		</include>
	</group>
	<group unless="$(arg online)">
		<param name="use_sim_time" value="true" />
		<include file="play_cloud.launch">
			<arg name="camera" value="$(arg camera)" />
			<arg name="filename" value="$(arg filename)" />
		</include>
	</group>

	<node name="drop1" pkg="topic_tools" type="drop" args="/$(arg camera)/depth_registered/points 2 3 /drop/points"/>

	<node name="preprocessor" pkg="binary_symlinks" type="preprocessor_node" output="screen"  args="--inputTopic=/drop/points --downsample=0.008 --removeOutliers=false --outlierRadius=0.016 --outlierMinK=0 --clusterMinSize=5 --zClipLow=-0.02"/> 

</launch>
